exchanges/okx/okx.go

```diff
@@ -0,0 +1,4441 @@
+		after, before, gridAlgoOrders, limit)
+}
+
+// GetGridAlgoOrderHistory retrieves list of grid algo orders with the complete data including the stoped orders.
+func (ok *Okx) GetGridAlgoOrderHistory(ctx context.Context, algoOrderType, algoID,
+	instrumentID, instrumentType,
+	after, before string, limit int64) ([]GridAlgoOrderResponse, error) {
+	return ok.getGridAlgoOrders(ctx, algoOrderType, algoID,
+		instrumentID, instrumentType,
+		after, before, gridAlgoOrdersHistory, limit)
+}
+
+// getGridAlgoOrderList retrieves list of grid algo orders with the complete data.
+func (ok *Okx) getGridAlgoOrders(ctx context.Context, algoOrderType, algoID,
+	instrumentID, instrumentType,
+	after, before, route string, limit int64) ([]GridAlgoOrderResponse, error) {
+	params := url.Values{}
+	algoOrderType = strings.ToLower(algoOrderType)
+	if algoOrderType != AlgoOrdTypeGrid && algoOrderType != AlgoOrdTypeContractGrid {
+		return nil, errMissingAlgoOrderType
+	}
+	params.Set("algoOrdType", algoOrderType)
+	if algoID != "" {
+		params.Set("algoId", algoID)
+	}
+	if instrumentID != "" {
+		params.Set("instId", instrumentID)
+	}
+	if instrumentType == okxInstTypeSpot || instrumentType == okxInstTypeMargin ||
+		instrumentType == okxInstTypeFutures || instrumentType == okxInstTypeSwap {
+		params.Set("instType", strings.ToUpper(instrumentType))
+	}
+	if after != "" {
+		params.Set("after", after)
+	}
+	if before != "" {
+		params.Set("before", before)
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	epl := getGridAlgoOrderListEPL
+	if route == gridAlgoOrdersHistory {
+		epl = getGridAlgoOrderHistoryEPL
+	}
+	var resp []GridAlgoOrderResponse
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, epl, http.MethodGet, common.EncodeURLValues(route, params), nil, &resp, true)
+}
+
+// GetGridAlgoOrderDetails retrieves grid algo order details
+func (ok *Okx) GetGridAlgoOrderDetails(ctx context.Context, algoOrderType, algoID string) (*GridAlgoOrderResponse, error) {
+	params := url.Values{}
+	if !(algoOrderType == AlgoOrdTypeGrid ||
+		algoOrderType == AlgoOrdTypeContractGrid) {
+		return nil, errMissingAlgoOrderType
+	}
+	if algoID == "" {
+		return nil, errMissingAlgoOrderID
+	}
+	params.Set("algoOrdType", algoOrderType)
+	params.Set("algoId", algoID)
+	var resp []GridAlgoOrderResponse
+	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getGridAlgoOrderDetailsEPL, http.MethodGet, common.EncodeURLValues(gridOrdersAlgoDetails, params), nil, &resp, true)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp) == 1 {
+		return &resp[0], nil
+	}
+	return nil, errNoValidResponseFromServer
+}
+
+// GetGridAlgoSubOrders retrieves grid algo sub orders
+func (ok *Okx) GetGridAlgoSubOrders(ctx context.Context, algoOrderType, algoID, subOrderType, groupID, after, before string, limit int64) ([]GridAlgoOrderResponse, error) {
+	params := url.Values{}
+	if !(algoOrderType == AlgoOrdTypeGrid ||
+		algoOrderType == AlgoOrdTypeContractGrid) {
```

Comment:
```suggestion
	if algoOrderType != AlgoOrdTypeGrid && algoOrderType != AlgoOrdTypeContractGrid {
```
The error is not missing if algoType is populated. Please update error